File: Semester 1/Class09_Computer_Contest_Lv_2_Notes_and_Homework_Nov_06-07/lvl2_hw_sol/lvl2_slot_machine.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    int quarters, first, second, third, total=0, time1=0, time2=0, time3=0;
    cin >> quarters >> time1 >> time2 >> time3;
    while(quarters > 0){
        time1++; quarters-=1; total++;
        if (time1%35 == 0){
            quarters+=30;
        }
        if (quarters <= 0){
            break;
        }
        time2++; quarters-=1; total++;
        if (time2%100 == 0){
            quarters+=60;
        }
        if (quarters <= 0){
            break;
        }
        time3++; quarters-=1; total++;
        if (time3%10 == 0){
            quarters+=9;
        }
    }
    cout << "Martha plays " << total << " times before going broke.";
    return 0;
}

File: Semester 1/Class09_Computer_Contest_Lv_2_Notes_and_Homework_Nov_06-07/lvl2_hw_sol/lvl2_slot_machine.java
import java.util.*;
import java.io.*;
public class slot_machine {
	public static void main(String[] args){
		Scanner in = new Scanner(System.in);
		int n = in.nextInt(), m1 = in.nextInt(), m2 = in.nextInt(), m3 = in.nextInt();
		int count = 0;
		while(true) {
			n--; m1++; count++;
			if(m1 == 35) {
				n += 30; m1 = 0;
			}
			if(n == 0) break;
			n--; m2++; count++;
			if(m2 == 100) {
				n += 60; m2 = 0;
			}
			if(n == 0) break;
			n--; m3++; count++;
			if(m3 == 10) {
				n += 9; m3 = 0;
			}
			if(n == 0) break;
		}
		System.out.println("Martha plays "+count+" times before going broke.");
	}
}